File: Programação Orientada a Objetos/Trabalho Prático 2/src/main/java/main/Chacara.java
package main;

public class Chacara extends Imovel {
    //Características específicas da chácara
    private boolean salao_festa;
    private boolean salao_jogos;
    private boolean campo_futebol;
    private boolean churrasqueira;
    private boolean piscina;

    //Construtor
    public Chacara (int i, float v, String p, String r, String b, String c, int n, int q, int ba, boolean sf, boolean sj, boolean cf, boolean ch, boolean pi) {
        super(i, v, p, r, b, c, n, q, ba);
        salao_festa = sf;
        salao_jogos = sj;
        campo_futebol = cf;
        churrasqueira = ch;
        piscina = pi;
    }

    //Método do printar
    @Override
    public void print() {
        super.print();
        if(piscina == true)
            System.out.printf("Tem piscina\n\n");
        else
            System.out.println("Não tem piscina\n\n");
    }

    //Getters e Setters
    public boolean getSalaoFesta() {
        return salao_festa;
    }

    public void setSalaoFesta(boolean sf){
        salao_festa = sf;
    }

    public boolean getSalaoJogos() {
        return salao_jogos;
    }

    public void setSalaoJogos(boolean sj){
        salao_jogos = sj;
    }

    public boolean getCampoFutebol() {
        return campo_futebol;
    }

    public void setCampoFutebol(boolean cf){
        campo_futebol = cf;
    }

    public boolean getChurrasqueira() {
        return churrasqueira;
    }

    public void setChurrasqueira(boolean c){
        churrasqueira = c;
    }

    public boolean getPiscina() {
        return piscina;
    }

    public void setPiscina(boolean p){
        piscina = p;
    }
}

File: Programação Orientada a Objetos/Trabalho Prático 1/chacara.cpp
#include "chacara.h"
#include "imovel.h"

using namespace std;

//Construtor
Chacara::Chacara(int i, float v, std::string p, std::string r, std::string b, std::string c, int n, int q, int ba, bool sf, bool sj, bool cf, bool ch, bool pi) : Imovel(i, v, p, r, b, c, n, q, ba){
    salao_festa = sf;
    salao_jogos = sj;
    campo_futebol = cf;
    churrasqueira = ch;
    piscina = pi;
}

//Overload do método print
std::ostream& Chacara::print(std::ostream& os) const{
    os  << "----\n"
        << "\t" << this->getProprietario() << "\n"
        << "\t" << this->getValor() << "\n"
        << "\t" << this->getQuartos() << "\n"
        << "\t" << this->getRua() << "\n"
        << "\t" << this->getBairro() << "\n"
        << "\t" << this->getCidade() << "\n"
        << "\t" << ((this->getPiscina()) ? "Possui piscina" : "Não possui piscina") << "\n"
        << "----\n";
    return os;
}

//Getters e Setters
bool Chacara::getSalaoFesta() const{
    return salao_festa;
}

void Chacara::setSalaoFesta(bool sf){
    salao_festa = sf;
}

bool Chacara::getSalaoJogos() const{
    return salao_jogos;
}

void Chacara::setSalaoJogos(bool sj){
    salao_jogos = sj;
}
 
bool Chacara::getCampoFutebol() const{
    return campo_futebol;
}

void Chacara::setCampoFutebol(bool cf){
    campo_futebol = cf;
}

bool Chacara::getChurrasqueira() const{
    return churrasqueira;
}

void Chacara::setChurrasqueira(bool c){
    churrasqueira = c;
}

bool Chacara::getPiscina() const{
    return piscina;
}

void Chacara::setPiscina(bool p){
    piscina = p;
}